applications/FemToDemApplication/custom_processes/extend_pressure_condition_process.cpp

```diff
@@ -39,6 +39,72 @@ void ExtendPressureConditionProcess<TDim>::GetMaximumConditionIdOnSubmodelPart(
     }
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+template <>
+void ExtendPressureConditionProcess<3>::CreatePressureLoads(
+    const int Id1,
+    const int Id2,
+    const int Id3,
+	ModelPart::ElementsContainerType::ptr_iterator itElem,
+	ModelPart& rSubModelPart,
+    ModelPart::PropertiesType::Pointer pProperties,
+    int& rMaximumConditionId
+    )
+{
+    auto& r_geom = (*itElem)->GetGeometry();
+    std::vector<IndexType> condition_nodes_id(3);
+    condition_nodes_id[0] = r_geom[Id1].Id();
+    condition_nodes_id[1] = r_geom[Id2].Id();
+    condition_nodes_id[2] = r_geom[Id3].Id();
+    rMaximumConditionId++;
+
+    // Adding the nodes to the SubModelPart
+	rSubModelPart.AddNode(mrModelPart.pGetNode(r_geom[Id1].Id()));
```

Comment:
I am quite sure that this is a Golberg machine. If the node already exists on the root modelpart it is enough with providing the Id, you are accessing to the database to the add it again  

https://en.wikipedia.org/wiki/Rube_Goldberg_machine

![imagen](https://user-images.githubusercontent.com/19991680/58466316-b4ec1300-8139-11e9-874d-c10a038edf9e.png)

Reply:
Im accessing the model part for adding that node to one SubModel, it is not the same I think :)

Reply:
The node is defined on the root model part, try to use the Id, is faster